### PR TITLE
Ajax loading of relations

### PR DESCRIPTION
**Pull request by zedsh**
Many related models make the interface very slow. Now this is fixed.
This commit uses the ajax-load of component select2 and adds one route for processing links.

### CODE REVIEW

File: src/Http/Controllers/VoyagerBaseController.php
Comment (fletch3555) [inline src/Http/Controllers/VoyagerBaseController.php]: Please remove commented code

File: src/Http/Controllers/VoyagerBaseController.php
Comment (fletch3555) [inline src/Http/Controllers/VoyagerBaseController.php:522]: I realize there are a lot that don't exist, but please add doc blocks to any new or modified methods in your PRs.  This will help improve the code moving forward.

File: src/Http/Controllers/VoyagerBaseController.php
Comment (fletch3555) [inline src/Http/Controllers/VoyagerBaseController.php]: Remove comment

File: src/Http/Controllers/VoyagerBaseController.php
Comment (fletch3555) [inline src/Http/Controllers/VoyagerBaseController.php:535]: `$details` doesn't seem to be used anywhere else.  Can we combine these 2 lines into one?

File: resources/views/formfields/relationship.blade.php
Comment (fletch3555) [inline resources/views/formfields/relationship.blade.php:35]: Please watch your tabs vs spaces.  This now looks ridiculous because you introduced tabs when everything else uses spaces.

**Review by fletch3555 — CHANGES_REQUESTED** [review]
Please remove the `mix.js` and `package-lock.json` files as they're not needed and have no bearing on this change.

Also, please undo your changes to `app.css` and `app.js`, as those will be recompiled automatically after this gets merged.  If everyone committed these files, everyone would have perpetual merge-conflicts, which is how it was before we implemented this process.

Finally, please add some kind of testing for this.  It's a fairly significant change from how it's currently working.

File: resources/views/formfields/relationship.blade.php
Comment (fletch3555) [inline resources/views/formfields/relationship.blade.php:74]: Please remove commented code.  It shouldn't be in the repo.

File: resources/views/formfields/relationship.blade.php
Comment (fletch3555) [inline resources/views/formfields/relationship.blade.php:126]: Same

**Review by fletch3555 — CHANGES_REQUESTED** [review]

**Comment by Rogue85** [comment]
Merge this PR please. It`s very necessary.

**Comment by fletch3555** [comment]
This has pending requested changes, si we can't merge it

**Comment by vince83110** [comment]
@zedsh Could you please do what fletch3555 asked you to change ? You only have to refactor your commit. Your PR is very important, I tried to to get your code, but I can't run the asset compilation, I miss something I guess...

**Comment by zedsh** [comment]
I'll try to do it this week.

**Comment by zedsh** [comment]
I made changes and wrote a small test.

**Comment by emptynick** [comment]
Superseeded by https://github.com/the-control-group/voyager/pull/3586

**Pull request closed without merging**
